op-service/rethdb-reader/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+#![doc = include_str!("../README.md")]
+
+use receipts::{read_receipts_inner, ReceiptsResult};
+use std::os::raw::c_char;
+
+mod receipts;
+
+/// Read the receipts for a blockhash from the RETH database directly.
+///
+/// # Safety
+/// - All possible nil pointer dereferences are checked, and the function will return a
+///   failing [ReceiptsResult] if any are found.
+#[no_mangle]
+pub unsafe extern "C" fn read_receipts(
```

Comment:
I'd prefix this symbol and `free_string` with `rethdb_reader_` or something. `read_receipts` is too generic.